modules/apigee_edge_teams/src/Form/TeamContextSwitcherForm.php

```diff
@@ -0,0 +1,155 @@
+
+/**
+ * Defines a form for switching team.
+ */
+class TeamContextSwitcherForm extends FormBase implements ContainerInjectionInterface {
+
+  /**
+   * The current route match.
+   *
+   * @var \Drupal\Core\Routing\RouteMatchInterface
+   */
+  protected $routeMatch;
+
+  /**
+   * The entity type manager.
+   *
+   * @var \Drupal\Core\Entity\EntityTypeManagerInterface
+   */
+  protected $entityTypeManager;
+
+  /**
+   * TeamContextSwitcherForm constructor.
+   *
+   * @param \Drupal\Core\Routing\RouteMatchInterface $route_match
+   *   The current route match.
+   * @param \Drupal\Core\Entity\EntityTypeManagerInterface $entity_type_manager
+   *   The entity type manager.
+   */
+  public function __construct(RouteMatchInterface $route_match, EntityTypeManagerInterface $entity_type_manager) {
+    $this->routeMatch = $route_match;
+    $this->entityTypeManager = $entity_type_manager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container) {
+    return new static(
+      $container->get('current_route_match'),
+      $container->get('entity_type.manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'apigee_edge_teams_context_switcher_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state, array $teams = []) {
+    $form_state->set('teams', $teams);
+
+    /** @var \Drupal\apigee_edge_teams\Entity\TeamInterface $team */
+    $options = [];
+    foreach ($teams as $id => $team) {
+      $options[$id] = $team->label();
+    }
+
+    // Get the current team from the route to use as default value.
+    /** @var \Drupal\apigee_edge_teams\Entity\TeamInterface $current_team */
+    $current_team = $this->routeMatch->getParameter('team') ?? NULL;
+
+    $title = (string) $this->t('Select @name', [
+      '@name' => $this->entityTypeManager->getDefinition('team')
+        ->getLowercaseLabel(),
+    ]);
+
+    $form['wrapper'] = [
+      '#type' => 'container',
+      '#attributes' => [
+        'class' => [
+          'container-inline',
+        ],
+      ],
+    ];
+
+    $form['wrapper']['team_id'] = [
+      '#title' => $title,
+      '#title_display' => 'invisible',
+      '#type' => 'select',
+      '#required' => TRUE,
+      '#options' => $options,
+      '#empty_option' => $title,
+      '#default_value' => $current_team ? $current_team->id() : NULL,
+    ];
+
+    $form['wrapper']['actions'] = [
+      '#type' => 'actions',
+    ];
+
+    $form['wrapper']['actions']['submit'] = [
+      '#type' => 'submit',
+      '#button_type' => 'primary',
+      '#value' => $this->t('Go'),
+    ];
+
+    return $form;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function submitForm(array &$form, FormStateInterface $form_state) {
+    /** @var \Drupal\apigee_edge_teams\Entity\TeamInterface $team */
+    if (($team_id = $form_state->getValue('team_id'))
```

Comment:
The indention of this if condition looks inappropriate, probably caused all these conditions checks...

Why *all* these sanity checks are needed? A form in Drupal cannot be submitted with a value that was not displayed there originally so `(isset($teams[$team_id]) && ($team = $teams[$team_id])` is unnnessary and form field is required therefore `$team_id = $form_state->getValue('team_id'))` is also redundant.